### ISSUE TITLE
Transform nullable columns 

### ISSUE THREAD

**Issue by KEALHOVIK** [kotlin-orm/ktorm#492]
If nullable properties are allowed in Entity and databases overall support null fields, **why wouldn't transform method be able to accept null from fromUnderlyingValue**.
IDK maybe i'm doing something wrong but i have serializable FloatRange class and as a field for it isn't essential i would prefer to get null on failed serialization. As it is now i can't find any other option but use some default value or split range into two columns for min and max.

I use it like so:
```
val range = varchar("range").transform(
     { FloatRange.parseFromString(it) ?: FloatRange(0f, 0f) }, { it.toString() }
).bindTo { it.range }
```

**Comment by vincentlauvlwj** [comment]
According to the fail fast principle, it's better to throw an exception on failed deserialization, not to return null or a default value.

**Comment by KEALHOVIK** [comment]
> According to the fail fast principle, it's better to throw an exception on failed deserialization, not to return null or a default value.

Yes, i understand that. But failed deserialization doesn't mean my app should fail too. In this situation fail fast works in a way to relieve serializer of responsibility for bad input and thus make it simpler and more maintainable. Shifting this responsibility on my code.
Also, i could've check if the value is parseble before deserializing it, thus avoiding exception. Thing is, this module shouldn't expect perfect data and it has a responsibility to do something if this happens. So, if i could produce a default value for this field and field could store null, why couldn't this default value be null?

**Comment by vincentlauvlwj** [comment]
If you get a null value from the entity, how can you distinguish these two situations? 

- The value in the database is null.
- The value in the database is not null but deserialization fails.

I understand that you don't want to throw an exception, another solution is to define a special value that represents deserialization failures

```kotlin
val invalidRange = Range(Float.NaN, Float.NaN)
```

**Comment by KEALHOVIK** [comment]
> 

That is the point, this transform definition is responsible for representation of my data. I decided that in my use case i want the value to be null if deserialization fails other parts of my app don't need to distinguish why could it be null.
Yes i can make a range of NaN but in such case i push part of transformation responsibility to parts of code that gonna use this entity. Although i agree this is pretty good solution too.
